Pinta.Core/Classes/Palette.cs

```diff
@@ -181,6 +159,32 @@ public void Load (Gio.File file)
 		}
 	}
 
+	static (List<Color>? loadedColors, StringBuilder errors) LoadColors (Gio.File file)
+	{
+		var loader = PintaCore.PaletteFormats.GetFormatByFilename (file.GetDisplayName ())?.Loader;
+
+		if (loader != null)
+			return (loader.Load (file), new StringBuilder ());
```

Comment:
It might be nicer to just return a string instead of StringBuilder since that's more of an implementation detail? It feels a bit weird to create an empty one